hostinfo/misc.lua

```diff
@@ -14,109 +14,86 @@ See the License for the specific language governing permissions and
-
-  local obj = {}
-  fs.exists(filePath, function(err, file)
-    if err then
-      table.insert(errTable, string.format('File not found : { fs.exists erred: %s }', err))
-      return callback()
-    end
-    if file then
-      fs.readFile(filePath, function(err, data)
-
-        if err then
-          table.insert(errTable, string.format('File couldnt be read : { fs.readline erred: %s }', err))
-          return callback()
-        end
 
-        for line in data:gmatch("[^\r\n]+") do
-          local iscomment = string.match(line, '^#')
-          local isblank = string.len(line:gsub("%s+", "")) <= 0
-
-          if not iscomment and not isblank then
-            -- split the line and assign key vals
-            local iter = line:gmatch("%S+")
-            casterFunc(iter, obj, line)
-          end
-        end
+local function _execFileToStreams(command, args, options)
+  local stdout, stderr = LineEmitter:new(), LineEmitter:new()
+  local child = childProcess.spawn(command, args, options)
+  child.stdout:pipe(stdout)
+  child.stderr:pipe(stderr)
+  return child, stdout, stderr
+end
 
-        -- Flatten single entry objects
-        if #obj == 1 then obj = obj[1] end
-        -- Dont insert empty objects into the outTable
-        if next(obj) then table.insert(outTable, obj) end
 
-        return callback()
-      end)
-    else
-      table.insert(errTable, 'file not found')
-      return callback()
+local function run(command, arguments, options)
+  local stream = Stream:new()
+  local called, exitCode
+  called = 2
+  local function done()
+    called = called - 1
+    if called == 0 then
+      if exitCode ~= 0 then
+        stream:emit('error', 'Process exited with exit code ' .. exitCode)
+      end
+      stream:emit('end')
     end
-
-  end)
+  end
+  local function onClose(_exitCode)
+    exitCode = _exitCode
+    done()
+  end
+
+  if not options.env then options.env = process.env end
+  local child, stdout, stderr = _execFileToStreams(command, arguments, options)
+  child:once('close', onClose)
+  stdout:on('data', function(data) stream:emit('data', data) end):once('end', done)
+  stderr:on('data', function(data) stream:emit('error', data) end)
+  return stream
 end
 
-local function asyncSpawn(dataArr, spawnFunc, successFunc, finalCb)
-  -- Asynchronous spawn cps & gather data
-  local obj = {}
-  async.forEachLimit(dataArr, 5, function(datum, cb)
-    local function _successFunc(err, exitcode, data, stderr)
-      successFunc(data, obj, datum, exitcode)
-      return cb()
-    end
-    local cmd, args = spawnFunc(datum)
-    return execFileToBuffers(cmd, args, { env = process.env }, _successFunc)
-  end, function()
-    return finalCb(obj)
-  end)
-end
 
+local function getInfoByVendor(options)
+  local sysinfo = sigar:new():sysinfo()
+  local vendor = sysinfo.vendor:lower()
+  local name = sysinfo.name:lower()
+  if vendor == 'red hat' then vendor = 'rhel' end
```

Comment:
yes.  vendor returns "Red Hat" 
I added
```
  local sysinfo = sigar:new():sysinfo()
  local vendor = sysinfo.vendor:lower()
  local name = sysinfo.name:lower()
 print(name)
 print(vendor)
```
to a random hostinfo check and ran it, heres a screenie. Latest version of red hat we offer. 
![screen shot 2015-08-20 at 5 01 09 pm](https://cloud.githubusercontent.com/assets/3248143/9398550/1787747a-475d-11e5-900d-c0f4de7b2173.png)